### PR TITLE
Refactor and improvements

### PR DESCRIPTION
**Pull request by hbcarlos**
Moves the menu bar and status bar buttons to JupterLab.

drawio

### CODE REVIEW

**Comment by hbcarlos** [comment]
There is an issue with menus not showing up because commands are added when the document it's opened.

menu

**Comment by hbcarlos** [comment]
We need to add shortcuts to the command, not in the label.

Screenshot from 2021-03-15 17-12-27

**Comment by jtpio** [comment]
Thanks!

A quick question after looking at the screencast. Should the menu entries be separate? (`Diagram Arrange`, `Diagram Extras`) Or grouped under a single `Diagram` entry? (with submenus)

File: src/drawio/modulated.js [hunk 1]
```diff
@@ -90857,9 +90857,9 @@ EditorUi.compactUi = true;
 EditorUi.prototype.splitSize = (mxClient.IS_TOUCH || mxClient.IS_POINTER) ? 12 : 8;
 
 /**
- * Specifies the height of the menubar. Default is 34.
+ * Specifies the height of the menubar. Default is 34. MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
```
Comment (jtpio) [inline src/drawio/modulated.js:90860]: Looks like this could be removed?

```suggestion
 * Specifies the height of the menubar. Default is 34.
```

**Comment by jtpio** [comment]
Were all the files under `src/mxgraph/dotnet`, `src/mxgraph/docs` and similar coming from the source `mxGraph` repo?

If so we can indeed remove them :+1:

File: package.json [hunk 2]
```diff
@@ -44,13 +45,13 @@
...
-    "@jupyterlab/filebrowser": "^3.0.2",
-    "@jupyterlab/launcher": "^3.0.2",
-    "@jupyterlab/mainmenu": "^3.0.2",
+    "@jupyterlab/application": "^3.1.0-alpha.2",
+    "@jupyterlab/apputils": "^3.1.0-alpha.2",
+    "@jupyterlab/coreutils": "^5.1.0-alpha.2",
+    "@jupyterlab/docregistry": "^3.1.0-alpha.2",
+    "@jupyterlab/filebrowser": "^3.1.0-alpha.2",
+    "@jupyterlab/launcher": "^3.1.0-alpha.2",
+    "@jupyterlab/mainmenu": "^3.1.0-alpha.2",
```
Comment (jtpio) [inline package.json:54]: If we plan to merge this and release a new version for JupyterLab 3.0, should we keep the `^3.0.x` packages for now?

Or was there something needed from the `3.1.0-alpha.x` packages?

**Comment by hbcarlos** [comment]
I think so. I'll remove it too.

File: package.json [hunk 3]
```diff
@@ -68,6 +69,13 @@
     "rimraf": "^3.0.2",
     "typescript": "~3.9.3"
   },
+  "resolutions": {
+    "@lumino/widgets": "~1.18.0",
+    "@types/react": "~17.0.0",
+    "csstype": "~2.4.0",
+    "react": "~17.0.1",
+    "typestyle": "~2.0.4"
+  },
```
Comment (jtpio) [inline package.json:78]: Just to double check: are these resolutions still needed?

File: package.json (hunk 2 above)
Comment (hbcarlos) [reply package.json:54]: Nothing, I'll update the packages.

File: package.json (hunk 3 above)
Comment (hbcarlos) [reply package.json:78]: No, that was the test for the RTC, same on the review above.

**Comment by hbcarlos** [comment]
> A quick question after looking at the screencast. Should the menu entries be separate? (`Diagram Arrange`, `Diagram Extras`) Or grouped under a single `Diagram` entry? (with submenus)

Good question indeed. Would look much nicer if grouping all under `Diagram` but will have 3 levels of submenus.
Let me do a screencast to see how it looks like.

**Comment by hbcarlos** [comment]
> Thanks!
> 
> A quick question after looking at the screencast. Should the menu entries be separate? (`Diagram Arrange`, `Diagram Extras`) Or grouped under a single `Diagram` entry? (with submenus)

Yep, looks nicer.

menu-diagram

**Comment by jtpio** [comment]
Looks nice, thanks for sharing the screencast :+1:

**Comment by wolfv** [comment]
wow, this looks _really_ properly integrated now! :tada:

**Comment by hbcarlos** [comment]
Thanks Jeremy!

**Comment by jtpio** [comment]
Just tested this PR locally and it looks great!

The Binder for this branch:

[Binder](https://mybinder.org/v2/gh/hbcarlos/jupyterlab-drawio/rtc?urlpath=lab)

seems to be failing at the `pip check` step:

https://github.com/QuantStack/jupyterlab-drawio/blob/4a7753d0d70ef4170106843b3aae97163e5d0c2b/binder/postBuild#L30

image

**Comment by jtpio** [comment]
Another thing that could be addressed separately is theming support for the toolbar icons, for example with the dark theme:

image

Maybe that will mean recreating a set of icons and use a `LabIcon` to add the items to the toolbar.

**Comment by hbcarlos** [comment]
Thanks Jeremy. Can you open an issue to address this later?

**Comment by jtpio** [comment]
Sounds good :+1:

**Review by jtpio — APPROVED** [review]
Thanks Carlos for working on this!

**Comment by jtpio** [comment]
Let's merge this and see if we can iterate in follow-up PRs before cutting a new release.

**Pull request merged**